tests/org/intellij/grammar/BnfGeneratorTest.java

```diff
@@ -24,6 +24,10 @@ public BnfGeneratorTest() {
   public void testTokenChoice() throws Exception { doGenTest(true); }
   public void testTokenChoiceNoSets() throws Exception { doGenTest(true); }
   public void testStub() throws Exception { doGenTest(true); }
+
+  public void testStubFallback() throws Exception {
+    doGenTest(true);
+  }
```

Comment:
could you also add a test for the case where stub fallback is set, but the actual stub class exists, so the fallback is not used for generation